hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -257,6 +260,7 @@ private void addInsertAction(AbstractEntityInsertAction insert) {
 			// TODO: find out why this is necessary
 			LOG.tracev( "Executing inserts before finding non-nullable transient entities for early insert: [{0}]", insert );
 			executeInserts();
+			earlyInsertQuerySpaces = insert.getPersister().getQuerySpaces();
```

Comment:
The root cause seems to be `EntityIdentityInsertAction` doesn't leave trace of the table to be updated. I made some exception for it to solve the issue.